src/oop/static_modifier/hard/q2/MainTest.java

```diff
@@ -0,0 +1,146 @@
+        _assertEquals(value1, testClass.getFieldValue(checkupInstance, attributeName), String.format(wrongValueMessage, className, attributeName));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void lightbulbClassHasCorrectSetVoltageMethod(String value1, int value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {value2, int.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String missingMethodMessage = "The %s class is missing the %s method. Make sure that the class contains the method and it is spelt correctly with the correct arguments as parameters.";
+        String wrongTypeMessage = "The %s method in the %s class has the wrong return type.";
+        String wrongModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setAttributeName, "does not correctly set the value of the", staticAttributeName, "attribute.");
+        assertTrue(testClass.hasMethod(setAttributeName, new Class<?>[]{int.class}),
+                String.format(missingMethodMessage, className, setAttributeName));
+        assertTrue(testClass.hasMethod(setAttributeName, new Class<?>[]{int.class}, Void.TYPE),
+                String.format(wrongTypeMessage, setAttributeName, className));
+        assertTrue(testClass.hasMethod(setAttributeName, new Class<?>[]{int.class}, Void.TYPE, new String[]{"public"}),
+                String.format(wrongModifierMessage, setAttributeName, className));
+        testClass.callMethod(setAttributeName, setSizeArguments, setMethodModifiers, classInstance);
+        _assertEquals(value2, testClass.getFieldValue(null, staticAttributeName), incorrectSetMethodMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependent1")
+    public void discordClassHasCorrectToStringMethod(String value1, int value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        testClass.setFieldValue(null, value2, staticAttributeName);
+        String[] methodModifiers = {"public"};
+        String expected = "This lightbulb is a " + value1 + " light with " + value2 + " voltage.";
+        String incorrectToStringMessage = String.join(" ",
+                "Your", className, "toString method does not return the correct String.");
+        Object output = testClass.callMethod("toString", methodModifiers, classInstance);
+        _assertEquals(expected, output, incorrectToStringMessage);
+    }
+
+    @Test
+    @Tag("dependent1")
+    public void correctMainMethod() throws Throwable {
```

Comment:
I think this should be `correctTestClass`

Reply:
Also, we should add a specific error message when the main prints the wrong output that is related to this question specifically.